src/renderer/features/wallets/WalletSelect/model/wallet-select-model.ts

```diff
@@ -1,67 +1,72 @@
-import { createStore, combine, createEvent, sample } from 'effector';
+import { createStore, combine, createEvent, sample, createApi, attach, createEffect } from 'effector';
 import BigNumber from 'bignumber.js';
+import { once, previous } from 'patronum';
 
-import { includes, getRoundedValue, totalAmount, dictionary } from '@shared/lib/utils';
+import { getRoundedValue, totalAmount, dictionary } from '@shared/lib/utils';
 import { walletModel, walletUtils, accountUtils } from '@entities/wallet';
 import { currencyModel, priceProviderModel } from '@entities/price';
-import type { WalletFamily, Wallet, ID } from '@shared/core';
-import { WalletType } from '@shared/core';
+import type { Wallet, ID } from '@shared/core';
 import { networkModel } from '@entities/network';
 import { balanceModel } from '@entities/balance';
+import { storageService } from '@shared/api/storage';
+import { walletSelectUtils } from '../lib/wallet-select-utils';
+
+export type Callbacks = {
+  onClose: () => void;
+};
 
 const walletIdSet = createEvent<ID>();
-const walletIdCleared = createEvent();
-const clearData = createEvent();
 const queryChanged = createEvent<string>();
+const walletSelected = createEvent<ID>();
 
-const $walletId = createStore<ID | null>(null).reset(walletIdCleared);
-const $filterQuery = createStore<string>('').reset(clearData);
-const $isWalletChanged = createStore<boolean>(false).reset(clearData);
+const $callbacks = createStore<Callbacks | null>(null);
+const callbacksApi = createApi($callbacks, {
+  callbacksChanged: (state, props: Callbacks) => ({ ...state, ...props }),
+});
 
-const $walletForDetails = combine(
+const $walletId = createStore<ID | null>(null);
```

Comment:
Do we need `walletId` store here?
Now dataflow is `walletIdSet` fired from wallet menu -> $walletId -> $walletForDetails. 
We can get rid of `$walletId` store and change `walletIdSet` to `walletForDetailsSet` and 
```
sample({
  source: walletModel.$wallets,
  clock: walletForDetailsSet,
  fn: ({wallets, id}) => {
    ...
  }
```
and change ` walletIdCleared: $walletId.reinit` to  `walletForDetailsCleared: $walletForDetails.reinit`. This way we make our dataflow one step simplier

Reply:
As I remember it was this way, but when you change wallet name it's not updated in `wallet-select-model`
In order to update we need to watch 2 things: `$walletId` and `walletModel.$wallets` (which is actually changes)